File: src/dataset/javascript/rna-transcription_exercism-javascript-ThomasZumsteg/gemini/GeminiFlash_rna-transcription_v2.js
var DnaTranscriber = function() {};

DnaTranscriber.prototype.toRna = function(dna) {
    const dnaToRna = {
        'G': 'C',
        'C': 'G',
        'T': 'A',
        'A': 'U'
    };

    let rna = "";
    for (let i = 0; i < dna.length; i++) {
        const nucleotide = dna[i];
        rna += dnaToRna[nucleotide] || '';
    }
    return rna;
};

export default DnaTranscriber;

File: src/dataset/go/rna_transcription_Exercism_go_ThomasZumsteg/gemini/GeminiFlash_rna_transcription_v2.go
package strand

//TestVersion is the unit tests that these functions will pass
const TestVersion = 1

//dnaToRna maps a dna nucleotide to an rna nucleotide
var dnaToRna = map[rune]rune{
	'G': 'C',
	'C': 'G',
	'T': 'A',
	'A': 'U',
}

/*ToRna converts a string of dna to rna*/
func ToRna(dnaStrand string) string {
	rnaStrand := make([]rune, len(dnaStrand))
	for i, dna := range dnaStrand {
		rnaStrand[i] = dnaToRna[dna]
	}
	return string(rnaStrand)
}